api/src/test/java/org/openmrs/scheduler/tasks/AutoRetireUsersTaskTest.java

```diff
@@ -0,0 +1,191 @@
+			.stream()
+			.filter(role -> Objects.equals(role.getRole(), RoleConstants.SUPERUSER))
+			.findFirst();
+
+		adminRole.ifPresent(inactiveUser::removeRole);
+
+		userService.saveUser(inactiveUser);
+
+		Set<User> usersToRetire = autoRetireUsersTask.getUsersToRetire(userService);
+
+		assertThat(usersToRetire, hasItem(inactiveUser));
+	}
+
+	@Test
+	public void getUsersToRetire_shouldNotReturnUsersThatDoNotExceedInactivityThreshold() {
+		administrationService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS, TWO_DAYS_PROPERTY_VALUE));
+
+		User inactiveUser = getDefaultUser();
+		inactiveUser.setUserProperty(
+			OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP, 
+			String.valueOf(System.currentTimeMillis() - ONE_DAY_IN_MILLISECONDS)
+		);
+
+		userService.saveUser(inactiveUser);
+
+		Set<User> usersToRetire = autoRetireUsersTask.getUsersToRetire(userService);
+
+		assertThat(usersToRetire, not(contains(inactiveUser)));
+	}
+
+	@Test
+	public void getUsersToRetire_shouldNotReturnAlreadyRetiredUsers() {
+		administrationService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS, ONE_DAY_PROPERTY_VALUE));
+
+		User retiredUser = getDefaultUser();
+		retiredUser.setUserProperty(
+			OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP, 
+			String.valueOf(System.currentTimeMillis() - TWO_DAYS_IN_MILLISECONDS)
+		); 
+		
+		retiredUser.setRetired(true);
+		retiredUser.setRetireReason("User retired due to inactivity");
+
+		userService.saveUser(retiredUser);
+
+		Set<User> usersToRetire = autoRetireUsersTask.getUsersToRetire(userService);
+
+		assertThat(usersToRetire, not(contains(retiredUser)));
+	}
+
+	@Test
+	public void userInactivityExceedsThreshold_shouldReturnTrueIfInactivityExceedsThreshold() {
+		User user = getDefaultUser();
+		user.setUserProperty(
+			OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP, 
+			String.valueOf(System.currentTimeMillis() - ONE_DAY_AND_TWO_SECONDS_IN_MILLISECONDS)
+		);
+
+		Optional<Role> adminRole = user.getAllRoles()
+			.stream()
+			.filter(role -> Objects.equals(role.getRole(), RoleConstants.SUPERUSER))
+			.findFirst();
+
+		adminRole.ifPresent(user::removeRole);
+
+		userService.saveUser(user);
+
+		boolean result = autoRetireUsersTask.userInactivityExceedsDaysToRetire(user, ONE_DAY_IN_MILLISECONDS);
+		assertTrue(result);
+	}
+
+	@Test
+	public void userInactivityExceedsThreshold_shouldReturnFalseIfInactivityDoesNotExceedThreshold() {
+		User user = getDefaultUser();
+		user.setUserProperty(
+			OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP, 
+			String.valueOf(System.currentTimeMillis() - TWENTY_THREE_HOURS_IN_MILLISECONDS)
+		);
+
+		userService.saveUser(user);
+
+		boolean result = autoRetireUsersTask.userInactivityExceedsDaysToRetire(user, ONE_DAY_IN_MILLISECONDS);
+		assertFalse(result);
+	}
+
+	@Test
+	public void userInactivityExceedsDaysToRetire_shouldReturnTrueIfUserInactiveSinceCreation() {
```

Comment:
The method name has should return true but i see one of the return values being false.